circuit/program/src/data/plaintext/equal.rs

```diff
@@ -51,7 +62,18 @@ impl<A: Aleo> Equal<Self> for Plaintext<A> {
                 }
                 false => Boolean::constant(true),
             },
-            (Self::Literal(..), _) | (Self::Struct(..), _) => Boolean::constant(true),
+            (Self::Array(a, _), Self::Array(b, _)) => match a.len() == b.len() {
+                true => {
+                    // Recursively check each element for inequality.
+                    let mut not_equal = Boolean::constant(false);
+                    for (plaintext_a, plaintext_b) in a.iter().zip_eq(b.iter()) {
+                        not_equal |= plaintext_a.is_not_equal(plaintext_b);
+                    }
+                    not_equal
```

Comment:
```suggestion
                    if a.iter().zip_eq(b.iter()).any(|(plaintext_a, plaintext_b)| plaintext_a.is_not_equal(plaintext_b)) {
                        Boolean::constant(true)
                    } else {
                        Boolean::constant(false)
                    }
```

Reply:
ditto: for short-circuiting, unless `is_not_equal` has to be applied to each element regardless of the end result

Reply:
In a circuit context, we can't short-circuit since it would change determinism